app/models/manageiq/providers/ansible_tower/provider.rb

```diff
@@ -43,8 +43,10 @@ def verify_credentials(auth_type = nil, options = {})
     validity
   rescue Faraday::ConnectionFailed, Faraday::SSLError => err
     raise MiqException::MiqUnreachableError, err.message, err.backtrace
-  rescue Faraday::Error::ClientError => err
-    raise MiqException::MiqCommunicationsError, JSON.parse(err.message)['detail']
+  rescue Faraday::ClientError => err
+    message = JSON.parse(err.message)['detail'] rescue nil
```

Comment:
@bdunne @gmcculloug - I am struggling a bit with the inline `rescue nil`.

I think I'd rather see this issue fixed in the gem - though I understand that will cause issues in the middleware - since we'll have to add a bunch of one off `raise` `rescue` calls.  @mkanoor - Mentioned to me this morning that he was getting odd errors when the Tower instance ran out of licensing for the number of managed hosts.  It returned an error that was a symptom of the issue instead of a licensing error.  

I just think we'll have to handle more of these in the future - so the gem may as well pretty all of them up.